lib/mongo/tracing/open_telemetry/command_tracer.rb

```diff
@@ -0,0 +1,320 @@
+        #
+        # @return [ Hash ] connection span attributes.
+        def connection_attributes(connection)
+          {
+            'server.port' => connection.address.port,
+            'server.address' => connection.address.host,
+            'network.transport' => connection.transport.to_s,
+            'db.mongodb.server_connection_id' => connection.server.description.server_connection_id,
+            'db.mongodb.driver_connection_id' => connection.id
+          }
+        end
+
+        # Returns session and transaction attributes.
+        #
+        # @param message [ Mongo::Protocol::Message ] the command message.
+        #
+        # @return [ Hash ] session span attributes.
+        def session_attributes(message)
+          {
+            'db.mongodb.cursor_id' => cursor_id(message),
+            'db.mongodb.lsid' => lsid(message),
+            'db.mongodb.txn_number' => txn_number(message)
+          }
+        end
+
+        # Processes cursor context from the command result.
+        #
+        # @param result [ Object ] the command result.
+        # @param _cursor_id [ Integer | nil ] the cursor ID (unused).
+        # @param _context [ OpenTelemetry::Context ] the context (unused).
+        # @param span [ OpenTelemetry::Trace::Span ] the current span.
+        def process_cursor_context(result, _cursor_id, _context, span)
+          return unless result.has_cursor_id? && result.cursor_id.positive?
+
+          span.set_attribute('db.mongodb.cursor_id', result.cursor_id)
+        end
+
+        # Records error status code if the command failed.
+        #
+        # @param result [ Object ] the command result.
+        # @param span [ OpenTelemetry::Trace::Span ] the current span.
+        def maybe_trace_error(result, span)
+          return if result.successful?
+
+          span.set_attribute('db.response.status_code', result.error.code.to_s)
+          begin
+            result.validate!
+          rescue Mongo::Error::OperationFailure => e
+            span.record_exception(e)
+          end
+        end
+
+        # Generates a summary string for the query.
+        #
+        # @param message [ Mongo::Protocol::Message ] the command message.
+        #
+        # @return [ String ] summary in format "command_name db.collection" or "command_name db".
+        def query_summary(message)
+          if (coll_name = collection_name(message))
+            "#{command_name(message)} #{database(message)}.#{coll_name}"
+          else
+            "#{command_name(message)} #{database(message)}"
+          end
+        end
+
+        # Extracts the collection name from the command message.
+        #
+        # @param message [ Mongo::Protocol::Message ] the command message.
+        #
+        # @return [ String | nil ] the collection name, or nil if not applicable.
+        def collection_name(message)
+          case message.documents.first.keys.first
```

Comment:
Will this be stable? Are we guaranteed that the first key will always be the name of the command?

Also, perhaps it would be clearer to use the `#command_name` method here, which has the same logic but also does `#to_s`.